### PR TITLE
Add clip and skip (nodraw) texture support

### PR DESCRIPTION
**Pull request by EIREXE**
Adds clip and nodraw texture support, as a property to TBLoader

### CODE REVIEW

File: src/builder.cpp [hunk 1]
```diff
@@ -447,6 +448,10 @@ MeshInstance3D* Builder::build_entity_mesh(int idx, LMEntity& ent, Node3D* paren
 		// Create material
 		Ref<Material> material;
 
+		if (String(tex.name) == m_loader->get_skip_texture_name()) {
```
Comment (codecat) [inline src/builder.cpp:451]: It would probably be good to have a comment here explaining what this is doing (even if it's clear by looking at the code)

Comment (codecat) [reply src/builder.cpp:451]: Additionally, casting `tex.name` to a `String` might invoke a memory allocation & string copy for each texture, we can probably avoid that? (Doesn't `String` have a compare method against `const char*`?)

File: src/builder.cpp [hunk 2]
```diff
@@ -482,12 +487,15 @@ MeshInstance3D* Builder::build_entity_mesh(int idx, LMEntity& ent, Node3D* paren
...
 			}
 
-			// Add surface to mesh
-			add_surface_to_mesh(mesh, surf);
+			add_surface_to_mesh(collision_mesh, surf);
 
-			// Give mesh material
-			if (material != nullptr) {
-				mesh->surface_set_material(mesh->get_surface_count() - 1, material);
+			// Add surface to mesh
```
Comment (codecat) [inline src/builder.cpp:492]: Misplaced comment? These should ultimately have 2 different comments, one for adding the surface to the visible mesh, and one for adding the surface to the collision mesh.

File: src/builder.cpp [hunk 3]
```diff
@@ -482,12 +487,15 @@ MeshInstance3D* Builder::build_entity_mesh(int idx, LMEntity& ent, Node3D* paren
...
+			if (String(tex.name) != m_loader->get_clip_texture_name()) {
```
Comment (codecat) [inline src/builder.cpp:493]: It would be cleaner to do an `if` / `continue` here I think

File: src/builder.cpp [hunk 4]
```diff
@@ -501,14 +509,15 @@ MeshInstance3D* Builder::build_entity_mesh(int idx, LMEntity& ent, Node3D* paren
...
 	case ColliderType::Static:
-		switch (colshape) {
-		case ColliderShape::Convex: mesh_instance->create_multiple_convex_collisions(); break;
-		case ColliderShape::Concave: mesh_instance->create_trimesh_collision(); break;
-		}
+		StaticBody3D* static_body = memnew(StaticBody3D());
+		static_body->set_name(String(mesh_instance->get_name()) + "_col");
+		parent->add_child(static_body, true);
+		static_body->set_owner(m_loader->get_owner());
+		add_collider_from_mesh(static_body, collision_mesh, colshape);
```
Comment (codecat) [inline src/builder.cpp:520]: Does this change the behavior from `mesh_instance->create_multiple_convex_collisions()` or `mesh_instance->create_trimesh_collision()`?

File: src/tb_loader.cpp [hunk 5]
```diff
@@ -115,6 +121,22 @@ bool TBLoader::get_filter_nearest()
...
+}
+
+void TBLoader::set_skip_texture_name(String skip_texture_name) {
+	m_skip_texture_name = skip_texture_name;
+}
+
+String TBLoader::get_skip_texture_name() {
+	return m_skip_texture_name;
+}
+
```
Comment (codecat) [inline src/tb_loader.cpp:139]: `{` should be on a new line here. Also I think we can pass the strings as `const String&`, no?

**Review by codecat — COMMENTED** [review]
Thanks for the PR! I have some comments.

**Comment by EIREXE** [comment]
all fixed

**Pull request merged**